src/main/java/is/ru/tictactoe/Board.java

```diff
@@ -0,0 +1,84 @@
+package tictactoe.src.main.java.is.ru.tictactoe;
+
+public class Board {
+	
+	private final static int BOARD_SIZE = 3;
+	private final static int MAX_PLAYERS = 2;
+	private int[][] board;
+	private int moves;
+	
+	public Board() {
+		this.board = new int[BOARD_SIZE][BOARD_SIZE];
+		reset();
+	}
+	
+	private void reset() {
+		this.moves = 0;
+		for (int x = 0; x < BOARD_SIZE; x++) {
+			for (int y = 0; y < BOARD_SIZE; y++) {
+				this.board[x][y] = 0;
+			}
+		}
+	}
+	
+	private void validCell(int x, int y) {
+		if (0 > x || x >= BOARD_SIZE)	throw new java.lang.IndexOutOfBoundsException("Illegal move");
+		if (0 > y || y >= BOARD_SIZE)	throw new java.lang.IndexOutOfBoundsException("Illegal move");		
+	}
+	
+	private void validPlayer(int z) {
+		if (z < 0 || z >= MAX_PLAYERS)	throw new java.lang.IllegalArgumentException("Illegal player");
+	}
+	
+	public boolean resetBoard() {
+		reset();
+		return true;
+	}
+
+	public int getCell(int x, int y) {
+		validCell(x, y);
+		return board[x][y];
+	}
+	
+	public boolean makeMove(int x, int y, int z) {
+		validCell(x,y);
+		validPlayer(z);
+		if (board[x][y] == 0) {
+			board[x][y] = z;
+			return true;
+		}
+		return false;
+	}
+	
+	public int winner() {
+		// return 1 for player 1
+		// return 2 for player 2
+		// return -1 for stale mate
+		// return 0 for no winner - not finished
```

Comment:
Maybe have this return an enum instead of constants?

Something like:
```
public enum GameResult {
  PLAYER_1,
  PLAYER_2,
  STALE_MATE,
  GAME_IN_PROGRESS,
}
```

Or return 0,1,2(for stalemate, player1, player2) and throw an exception in the case where a game is still in progress, but supply a 'gameFinished()' function to allow the caller to find out if the game is done or not?

I dunno, I feel it's hard to interface with an API that might return this many values and you don't get compiler help as in the case when you switch an enum